Sources/SmithyHTTPAPI/HTTPRequest.swift

```diff
@@ -35,24 +36,27 @@ public final class HTTPRequest: RequestMessage, @unchecked Sendable {
     public var path: String { destination.path }
     public var queryItems: [URIQueryItem]? { destination.queryItems }
     public var trailingHeaders: Headers = Headers()
-    public var endpoint: Endpoint {
-        return Endpoint(uri: self.destination, headers: self.headers)
-    }
+    public var endpoint: Endpoint { .init(uri: destination, headers: headers) }
+    public internal(set) var signedAt: Date?
```

Comment:
`signedAt` property is added to `HTTPRequest`.  It is `nil` until the request is signed, then it is set with the `Date` that was used as "now" for signing.